src/frontend/apps/impress/src/features/docs/doc-table-content/components/DocSummary.tsx

```diff
@@ -0,0 +1,151 @@
+import { MAIN_LAYOUT_ID } from '@/layouts/conf';
+
+import { Heading } from './Heading';
+
+type DocSummaryProps = {
+  headings: HeadingBlock[];
+};
+
+export const DocSummary = ({ headings }: DocSummaryProps) => {
+  const { editor } = useEditorStore();
+
+  const [headingIdHighlight, setHeadingIdHighlight] = useState<string>();
+
+  const { t } = useTranslation();
+  const [isHover, setIsHover] = useState(false);
+
+  useEffect(() => {
+    const handleScroll = () => {
+      if (!headings) {
+        return;
+      }
+
+      for (const heading of headings) {
+        const elHeading = document.body.querySelector(
+          `.bn-block-outer[data-id="${heading.id}"] [data-content-type="heading"]:first-child`,
+        );
+
+        if (!elHeading) {
+          return;
+        }
+
+        const rect = elHeading.getBoundingClientRect();
+        const isVisible =
+          rect.top + rect.height >= 1 &&
+          rect.bottom <=
+            (window.innerHeight || document.documentElement.clientHeight);
+
+        if (isVisible) {
+          setHeadingIdHighlight(heading.id);
+          break;
+        }
+      }
+    };
+
+    document.getElementById(MAIN_LAYOUT_ID)?.addEventListener('scroll', () => {
+      setTimeout(() => {
+        handleScroll();
+      }, 300);
+    });
+
+    handleScroll();
+
+    return () => {
+      window.removeEventListener('scroll', handleScroll);
+    };
+  }, [headings, setHeadingIdHighlight]);
+
+  if (!editor) {
+    return null;
+  }
+
+  return (
+    <Box
+      onMouseEnter={() => {
+        setIsHover(true);
+        setTimeout(() => {
+          const element = document.getElementById(
+            `heading-${headingIdHighlight}`,
+          );
+
+          element?.scrollIntoView({
+            behavior: 'smooth',
+            inline: 'center',
+            block: 'center',
+          });
+        }, 250); // 300ms is the transition time of the box
+      }}
+      onMouseLeave={() => {
+        setIsHover(false);
+      }}
+      id="summaryContainer"
+      $effect="show"
+      $width="40px"
+      $height="40px"
+      $zIndex={1000}
+      $align="center"
+      $padding="xs"
+      $justify="center"
+      $css={css`
+        border: 1px solid #ccc;
+        border: 1px solid #ccc;
+        overflow-x: hidden;
+        overflow-y: hidden;
+        border-radius: var(--c--theme--spacings--3xs);
+        background: var(--c--theme--colors--greyscale-000);
+
+        &:hover {
+          overflow-y: auto;
+          display: flex;
+          flex-direction: column;
+          justify-content: flex-start;
+          align-items: flex-start;
+          gap: var(--c--theme--spacings--2xs);
+          width: 200px;
+          height: auto;
+          max-height: calc(100vh - 60px - 15vh);
+        }
+      `}
+    >
+      {!isHover && (
+        <Box $justify="center" $align="center">
+          <Icon iconName="list" $theme="primary" $variation="800" />
+        </Box>
```

Comment:
Maybe add a small margin ? It is very close from the scroll bar no?
![image](https://github.com/user-attachments/assets/40c21cc3-9e5b-4e76-bca3-2fb6566535b4)